src/main/java/net/modfest/scatteredshards/client/ShardToast.java

```diff
@@ -34,7 +35,7 @@ public Visibility draw(DrawContext graphics, ToastManager manager, long startTim
 		TextRenderer textRenderer = manager.getClient().textRenderer;
 		
 		if (shard == null) return Toast.Visibility.HIDE;
-		ShardType shardType = shard.getShardType();
+		ShardType shardType = ScatteredShardsAPI.getClientLibrary().shardTypes().get(shard.shardTypeId()).orElse(ShardType.MISSING);
```

Comment:
I think there should be a convenience method in the shard library interface that does this

Reply:
Weirdly, even though I agree, the above is way too awkward - this is the hardest one to comply with. What would such a convenience method look like?